src/app/asset-upload/asset-upload.component.ts

```diff
@@ -8,12 +8,39 @@ import {AssetUploadService} from './asset-upload.service';
 })
 export class AssetUploadComponent {
   file: File | null = null;
+  assetLink = '';
+  isSizeInvalid = false;
 
   constructor(private service: AssetUploadService) {}
 
+  assetLinkUpload(assetLink: string) {
+    // Fetch the asset and parse the response stream as a blob
+    const assetBlob = fetch(assetLink)
+      .then(response => response.blob())
+      .then(blob => {
+        const filename = assetLink.substring(assetLink.lastIndexOf('/'));
+        const assetFile = new File([blob], filename);
+        this.isSizeInvalid =
+          assetFile.type.includes('video') && assetFile.size > 4000000
+            ? true
+            : false;
+        !this.isSizeInvalid
+          ? this.service.updateAssets(assetLink, assetFile)
+          : this.service.updateAssets('', null);
+        document.getElementById('filename').innerHTML = '';
+      });
+  }
+
   onFileInput(fileInput: any) {
     this.file = fileInput.target.files[0];
     const assetSrc = URL.createObjectURL(this.file);
-    this.service.updateAssets(assetSrc, this.file);
+    this.isSizeInvalid =
```

Comment:
Can we create a helper function with the size check and share between link and file?

Reply:
Completed this in final touches PR